libflux/core/src/scanner/mod.rs

```diff
@@ -64,19 +68,19 @@ impl Scanner {
     pub fn new(data: CString) -> Self {
```

Comment:
If we're gonna drop all the `CString` usages in this patch (I had figured we'd defer them, but this is fine) we should also change this to take a `&str`, which simplifies the `Parser` consumption of `Scanner::new` as well.

Reply:
Yeah agree, the parser had turned the input into a C string I think because in prep for the scanner needing it as a C string, but that's no longer the case. So no sense in converting to a C string, unless we need to in order to scan a [u8] ?

Reply:
We can do `"myString".as_bytes()` to get a `[u8]` from a `str`, [as documented here](https://doc.rust-lang.org/std/primitive.str.html#method.as_bytes)